Core/Src/main.c

```diff
@@ -95,19 +96,61 @@ int main(void)
   MX_ADC_Init();
   MX_TIM16_Init();
   MX_USART1_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
 
   /* USER CODE BEGIN 2 */
-  debug("\n\nBootcamp starting up...");
-  debug("Compiled on %s at %s", __DATE__, __TIME__);
 
+  //start pwm
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  //define buffers and variables
+  uint8_t spi_recieve_buf[10];
+  uint8_t spi_transmit_buf[10];
+  uint16_t TRANSMIT_RECIEVE_SIZE = 3;
+  uint16_t adc_val = 0x0000;
+
+  uint16_t MAX_ADC_VAL = 4095;
+  uint16_t MIN_COMPARE_VAL = 3000; // 60000 * 0.05 = 3000... NOTE: 60000 is the period AND 3000 is also the difference between the max and min compare val: 6000 - 3000 = 3000
+
+  double compare_val = 0.0;
+
+  //populate transmit buffer with start and config bits
+  spi_transmit_buf[0] = 0x01; //start bit
+  spi_transmit_buf[1] = 0x80; //config bit for single mode, CH0
+  spi_transmit_buf[2] = 0x00; //these are don't care bits, just filling them in bec we are sending and recieving a total of 3 bytes
+
+  //ensure the CS line starts high
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
 
-  HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); //CS low
+	  HAL_SPI_TransmitReceive(&hspi1, &spi_transmit_buf, &spi_recieve_buf, TRANSMIT_RECIEVE_SIZE, 1000); //sending the start and config bits as well as receiving the conversion
```

Comment:
I believe this function does not require the addresses of the transmit and receive buffers. So the '&' character can be removed for spi_transmit_buf and spi_recieve_buf.

Reply:
This is the function prototype here:

HAL_StatusTypeDef HAL_SPI_TransmitReceive(SPI_HandleTypeDef *hspi, uint8_t *pTxData, uint8_t *pRxData, uint16_t Size, uint32_t Timeout);

Isn't the buffer a pointer?